x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/mock/test_providers/test_providers.tsx

```diff
@@ -75,21 +88,221 @@ export const TestProvidersComponent: React.FC<Props> = ({ children, isILMAvailab
+    openCreateCaseFlyout: jest.fn(),
+    patterns: ['auditbeat-*'],
+    setLastChecked: jest.fn(),
+    startDate: null,
+    theme: {
+      background: {
+        color: '#000',
+      },
+    },
+    baseTheme: {
+      background: {
+        color: '#000',
+      },
+    },
+    ilmPhases: ['hot', 'warm', 'unmanaged'],
+    selectedIlmPhaseOptions: [
+      {
+        label: 'hot',
+        value: 'hot',
+      },
+      {
+        label: 'warm',
+        value: 'warm',
+      },
+      {
+        label: 'unmanaged',
+        value: 'unmanaged',
+      },
+    ],
+    setSelectedIlmPhaseOptions: jest.fn(),
+    ...dataQualityContextProps,
+  };
+
+  const mergedResultsRollupContextProps = {
+    onCheckCompleted: jest.fn(),
+    patternIndexNames: {
+      'auditbeat-*': [
+        '.ds-auditbeat-8.6.1-2023.02.07-000001',
+        'auditbeat-custom-index-1',
+        'auditbeat-custom-empty-index-1',
+      ],
+    },
+    patternRollups: {
+      'auditbeat-*': auditbeatWithAllResults,
+    },
+    totalDocsCount: 19127,
+    totalIncompatible: 4,
+    totalIndices: 3,
+    totalIndicesChecked: 3,
+    totalSameFamily: 0,
+    totalSizeInBytes: 18820446,
+    updatePatternIndexNames: jest.fn(),
+    updatePatternRollup: jest.fn(),
+    ...resultsRollupContextProps,
+  };
+
+  const mergedIndicesCheckContextProps: UseIndicesCheckReturnValue = {
+    checkIndex: jest.fn(),
+    checkState: {},
+  };
+
+  for (const key of Object.keys(mergedResultsRollupContextProps.patternIndexNames)) {
+    for (const indexName of mergedResultsRollupContextProps.patternIndexNames[key]) {
+      const mappingsProperties = getMappingsProperties({
+        indexName,
+        indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
+      });
+      const unallowedValues = getUnallowedValues({
+        requestItems: getUnallowedValueRequestItems({
+          ecsMetadata: EcsFlatTyped,
+          indexName,
+        }),
+        searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
+      });
+      const partitionedFieldMetadata = getSortedPartitionedFieldMetadata({
+        ecsMetadata: EcsFlatTyped,
+        loadingMappings: false,
+        mappingsProperties,
+        unallowedValues,
+      });
+      mergedIndicesCheckContextProps.checkState[indexName] = {
+        isChecking: false,
+        isLoadingMappings: false,
+        isLoadingUnallowedValues: false,
+        indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
+        partitionedFieldMetadata,
+        searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
+        unallowedValues,
+        mappingsProperties,
+        genericError: null,
+        mappingsError: null,
+        unallowedValuesError: null,
+        isCheckComplete: true,
+      };
+    }
+  }
+
+  Object.assign(mergedIndicesCheckContextProps, indicesCheckContextProps);
```

Comment:
nit: Consider eliminating local mutation of `mergedIndicesCheckContextProps.state` and mutation via `Object.assign` with something like:

```typescript
  const mergedIndicesCheckContextProps: UseIndicesCheckReturnValue = {
    checkIndex: jest.fn(),
    checkState: Object.keys(
      mergedResultsRollupContextProps.patternIndexNames
    ).reduce<UseIndicesCheckCheckState>((acc, key) => {
      mergedResultsRollupContextProps.patternIndexNames[key].forEach((indexName) => {
        const mappingsProperties = getMappingsProperties({
          indexName,
          indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
        });

        const unallowedValues = getUnallowedValues({
          requestItems: getUnallowedValueRequestItems({
            ecsMetadata: EcsFlatTyped,
            indexName,
          }),
          searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
        });

        const partitionedFieldMetadata = getSortedPartitionedFieldMetadata({
          ecsMetadata: EcsFlatTyped,
          loadingMappings: false,
          mappingsProperties,
          unallowedValues,
        });

        acc[indexName] = {
          isChecking: false,
          isLoadingMappings: false,
          isLoadingUnallowedValues: false,
          indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
          partitionedFieldMetadata,
          searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
          unallowedValues,
          mappingsProperties,
          genericError: null,
          mappingsError: null,
          unallowedValuesError: null,
          isCheckComplete: true,
        };
      });

      return acc;
    }, {}),
    ...indicesCheckContextProps,
  };
```

Reply:
actually I decided to build on this and fully refactor this gigantic function let me know what you think. I pushed the changes, and thanks for looking through this.